libbpfgo.go

```diff
@@ -531,6 +531,45 @@ func CreateMap(mapType MapType, mapName string, keySize, valueSize, maxEntries i
 	}, nil
 }
 
+// GetMapByName gets an existing map
+// Can be used for populating BPF array of maps or hash of maps. Currently, the returned map can only be updated.
+func GetMapByName(name string) (*BPFMap, error) {
```

Comment:
The logic used in GetMapByName is also being worked on in #343. @aymericDD could you please hold this effort for a while? So you can rebase it on top of upcoming changes. Thank you.

Reply:
Yes, no problem 👍 We manage to not be dependent of the libbpfgo for now, but it will more clean to be able to use it for the future.

Reply:
@aymericDD we are going in a different direction when using map of maps. You can see more in https://github.com/aquasecurity/libbpfgo/pull/343#issuecomment-1660998211.

That said, for your GetMapByName() to be in the managed part of the lib, it must have the BPFModule as an argument and search the map in it.

Would that solve your use case?

Reply:
Thanks to let me know 👍 
FWIK: it does not solve my use case, because the eBPF map is loaded from an [eBPF tc filter](https://man7.org/linux/man-pages/man8/tc-bpf.8.html). If I try to load a `BPFModule` `bpfModule, err := bpf.NewModuleFromFile("myprgm.bpf.o")`, it will create a new map not related to the map of the eBPF tc filter. The idea of this function is really to get a map already instantiated from another program. I hope it makes sense 👍  

The correct behavior:
![image](https://github.com/aquasecurity/libbpfgo/assets/8859832/1732a4aa-5e7b-4385-a5c3-00e8c96d436e)

If I use the BPF module:
![image](https://github.com/aquasecurity/libbpfgo/assets/8859832/76285cd9-47d3-42c3-8620-89631403fedb)

Reply:
I see, you need access to maps instantiated by other origins (not managed by any modules loaded through conventional means).

This falls in the case we cannot return a `BPFMap` type that will fail at some point of execution (for example, through methods that need a reference to the libbpf module and/or map objects - `b.bpfMap`).

In my previous PR https://github.com/aquasecurity/libbpfgo/pull/343, which was a continuation of @kakkoyun's, I tried to work around this issue by putting checks on all BPFMap methods, but in hindsight we assumed that it's not the right approach.

We need to make this obvious to the libbpfgo consumer, so BFPMapOfMaps is a soon-to-be-coming type with its own possible (and necessary) methods.

Regarding your use case, I assume that we need to return another type of BPFMap than the managed one and handle all its methods without confusing the latter. `BPFMapUnmanaged`? I'm still terrible at naming things.

Reply:
Exactly 👌  My problem is the map is instantiated by another origin. The idea of having a new kind a map looks good to me 👍 For the name of the map it's up to you 😉